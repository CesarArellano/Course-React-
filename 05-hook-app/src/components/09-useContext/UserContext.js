import { createContext } from 'react';

//Higher Order Component
export const UserContext = createContext(null);
