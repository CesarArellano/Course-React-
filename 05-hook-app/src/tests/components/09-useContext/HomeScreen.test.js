import { shallow, mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Testing with HomeScreen', () => {
  const user = {
    name: 'CÃ©sar',
    email: 'cesarmauricio.arellano@gmail.com'
  };

  test('should show correctly', () => {
    const wrapper = mount(
      <UserContext.Provider value = {{
        user
      }}>

        <HomeScreen />
      </UserContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  })
  
})
