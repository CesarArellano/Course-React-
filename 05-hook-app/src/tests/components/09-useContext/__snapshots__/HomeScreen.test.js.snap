// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with HomeScreen should show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "CÃ©sar",
   "email": "cesarmauricio.arellano@gmail.com"
}
  </pre>
</div>
`;
